### ISSUE TITLE
Attachments are duplicated in app/upload/users/.../message_attachments

### ISSUE THREAD

**Issue by p4pr8n** [chamilo/chamilo-lms#2841]
Using  Chamilo for a  while I would be happy to see it using less disk space.
I have found that folder ../app/upload/users/../message_attachments could use Gigas of space.
After making a md5sum on all the files I found that a lot of these files were identical.
Could it be possible to make a better management of these files ?
The workaround is to use hardlink to make the job but it's not very sexy.

Thanks

P.

**Comment by p4pr8n** [comment]
Example in the real world

```
.../app/upload/users/.../message_attachments# hardlink -n *
Mode:     dry-run
Files:    532
Linked:   369 files
Compared: 0 xattrs
Compared: 369 files
Saved:    997,98 MiB
Duration: 3,91 seconds
```

**Comment by ywarnier** [comment]
Hi @p4pr8n 
Chamilo 2.0 has a considerable part of its goals revolving around enabling sharing of objects vs copying them. However, this will not apply (at least in version 2.0, maybe the next) to files in app/upload/users/, which are supposed to be... personal.
We will try to improve it progressively in the future but this is the typical well-wishing feature request that is impossible to implement with the level of financing that we have at the moment.
It's not a lack of will, but there are other priorities that have overall better chances to benefit more people.

**Comment by p4pr8n** [comment]
I agree but folder message_attachments has something to do with mail application.
Why are attached files duplicated so many times ?
I'll try to get more details.

**Comment by ywarnier** [comment]
No idea. Maybe it's a common image in the Chamilo icons library or something. Maybe it's a hacked movie ;-)

**Comment by jmontoyaa** [comment]
Also the message attachments that are deleted are marked with the "_DELETED_" string in that folder.

**Comment by p4pr8n** [comment]
I have more details.
When creating an announcement with an attached file, the file is duplicated for each recipient in the sender's folder.

**Comment by ywarnier** [comment]
Just a heads up: we have verified this issue and, as a result and as a temporary hack, we have added the `$_configuration['disable_announcement_attachment']` variable in configuration.php (for 1.11.12) to prevent teachers from attaching items to the announcements :-)

This is only temporary until we develop a way to only keep one version of the file on disk.

**Comment by ywarnier** [comment]
This feature is implemented in Chamilo 2.0. A file is only uploaded once and then linked from any message that included this file.
